File: digae_backend/src/main/java/com/digae/sga/dto/response/BitacoraResiduoResponseDTO.java
package com.digae.sga.dto.response;

import com.digae.sga.entity.enums.ClasificacionResiduo;
import com.digae.sga.dto.request.ResiduoDTO;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * DTO de response para BitacoraResiduo.
 * Incluye el nombre del usuario que registró la bitácora.
 */
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class BitacoraResiduoResponseDTO {

    private Long id;
    private LocalDate fechaRegistro;
    private java.util.List<ResiduoDTO> residuos;
    private String areaGeneradora;
    private String empresaRecolectora;
    private String firmaBase64;
    private String observaciones;
    private Long usuarioId;
    private String usuarioNombreCompleto;
    private LocalDateTime creadoEn;
}

File: digae_android/app/src/main/java/com/grupo7poo2/digae/network/dto/BitacoraResiduoResponseDTO.kt
package com.grupo7poo2.digae.network.dto

import com.google.gson.annotations.SerializedName

data class BitacoraResiduoResponseDTO(
    @SerializedName("id")
    val id: Long,
    @SerializedName("fechaRegistro")
    val fechaRegistro: String,
    @SerializedName("areaGeneradora")
    val areaGeneradora: String,
    @SerializedName("empresaRecolectora")
    val empresaRecolectora: String?,
    @SerializedName("observaciones")
    val observaciones: String?,
    @SerializedName("usuarioId")
    val usuarioId: Long,
    @SerializedName("usuarioNombreCompleto")
    val usuarioNombreCompleto: String?,
    @SerializedName("creadoEn")
    val creadoEn: String?,
    @SerializedName("residuos")
    val residuos: List<ResiduoDTO>?
)
